### DISCUSSION TITLE
Unflattening and property type initialization

### DISCUSSION THREAD

**Discussion by forrestab** [riok/mapperly#307] [Q&A] [answered]
This could just be my misunderstanding, but I was looking at the tests to make sure the syntax for unflattening was simply the reversal of flattening and noticed that the target property type is not initialized. Is it that the library expects you to initialize all property types that aren't nullable in the default constructor?

Taking this [test](https://github.com/riok/mapperly/blob/main/test/Riok.Mapperly.Tests/Mapping/ObjectPropertyFlatteningTest.cs#L164) as an example, in the generated code I would have expected `target.Value = new();` just before setting the id.

Thanks.

**Comment by latonz** [comment, marked as answer]
If nullable reference types are enabled for Mapperly, Mapperly does not initialize proeprties marked as non-nullable.
If you look for example at these two test: https://github.com/riok/mapperly/blob/01af24fc9abd8f9c33aadf7940b69cade45704ec/test/Riok.Mapperly.Tests/Mapping/ObjectPropertyFlatteningTest.cs#L162-L202
Both have nullable reference types enabled. In the `ManualUnflattenedProperty` test the `Value` property of the class `B` **is not** nullable and therefore does *not* get initialized by Mapperly. In the `ManualUnflattenedPropertyNullablePath` test the `Value` property of the class `B` **is** nullable and therefore gets initialized by Mapperly.
